tests/src/core/testqgslayoutmap.cpp

```diff
@@ -132,6 +135,17 @@ void TestQgsLayoutMap::init()
 #endif
 }
 
+void TestQgsLayoutMap::construct()
+{
+  QgsLayout l( QgsProject::instance( ) );
+  QgsLayoutItemMap mi( &l );
+  QCOMPARE( mi.type(), QgsLayoutItemRegistry::LayoutMap );
+  QVERIFY( mi.rect().isNull() ); // is this correct to expect ?
+  QVERIFY( mi.extent().isNull() ); // is this correct to expect ?
+  QCOMPARE( mi.scale(), 0 ); // is this correct ?
```

Comment:
It's important to know what to expect because the zoomToExtent  functionality depends on the current extent so in order to test that I need to know what to expect as a current extent when the item map is constructed. Is it supposed to depend on the layout instance ? The constructor documentation doesn't say anything about what to expect as the initial state, is that to be considered undefined behavior ? I think it's good for tests to fail if defaults are changed, as users and other parts of the code are likely depending on defaults ?

Reply:
Maybe we should change defaults now if we believe the current defaults are not good. Like, should scale default to 1 instead of 0 ?

Reply:
> considered undefined behavior 

Yes, I think so. There's a LOT of setup you need to do to get a map item into a valid state, eg:

- set the position and size on the page
- set the map extent
- set the map crs

None of these are things we can set any meaningful defaults for, and it's all properties which aren't available through the layout at time of map item creation.

Reply:
> Yes, I think so. There's a LOT of setup you need to do to get a map item into a valid state, eg:

Would it make sense to expose an isValidState() method and throw an exception from method trying to access properties that would only make sense in "valid state" ? Also I see other tests ( like ::rendere() ) using QgsLayout.initializeDefaults(), is that what currently turns the state of the ItemMap valid ?

Reply:
> Would it make sense to expose an isValidState() method

Yes! :100: 

> and throw an exception from method trying to access properties that would only make sense in "valid state" 

That'd be an API break -- it'd have to wait for 4.0

>QgsLayout.initializeDefaults(), is that what currently turns the state of the ItemMap valid ?

"Initializes an empty layout, e.g. by adding a default page to the layout. This should be called after creating
a new layout."

Reply:
Ok for now I've rebased against current master branch and removed the rect and scale checks from the constructor test, leaving the one for current extent, which is what I'm using in ::zoomToExtent test. The isValidState() may be better to do in a separate PR